registry/storage/driver/rados/rados.go

```diff
@@ -0,0 +1,627 @@
+	}
+
+	// Delete old reference
+	err = d.DeleteOid(sourcePath)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Delete recursively deletes all objects stored at "path" and its subpaths.
+func (d *driver) Delete(path string) error {
+	// Get oid
+	oid, err := d.GetOid(path)
+
+	if err != nil {
+		return err
+	}
+
+	// Deleting virtual directory
+	if oid == "" {
+		objects, err := d.ListDirectoryOid(path)
+		if err != nil {
+			return err
+		}
+
+		for object := range objects {
+			err = d.Delete(filepath.Join(path, object))
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		// Delete object chunks
+		totalSize, err := d.GetXattrTotalSize(oid)
+
+		if err != nil {
+			return err
+		}
+
+		for offset := uint64(0); offset < totalSize; offset += d.chunksize {
+			chunkName, _ := d.GetChunkNameFromOffset(oid, offset)
+
+			err = d.Ioctx.Delete(chunkName)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Delete reference
+		err = d.DeleteOid(path)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *driver) URLFor(path string, options map[string]interface{}) (string, error) {
+	return "", storagedriver.ErrUnsupportedMethod
+}
+
+// Generate a blob identifier
+func (d *driver) GenerateOid() string {
+	return objectBlobPrefix + uuid.New()
+}
+
+// Reference a object and its hierarchy
+func (d *driver) PutOid(path string, oid string) error {
+	directory := filepath.Dir(path)
+	base := filepath.Base(path)
+	createParentReference := true
+
+	// After creating this reference, skip the parents referencing since the
+	// hierarchy already exists
+	if oid == "" {
+		firstReference, err := d.Ioctx.GetOmapValues(directory, "", "", 1)
+		if (err == nil) && (len(firstReference) > 0) {
+			createParentReference = false
+		}
+	}
+
+	oids := map[string][]byte{
+		base: []byte(oid),
+	}
+
+	// Reference object
+	err := d.Ioctx.SetOmap(directory, oids)
+	if err != nil {
+		return err
+	}
+
+	// Esure parent virtual directories
+	if createParentReference && directory != "/" {
+		return d.PutOid(directory, "")
+	}
+
+	return nil
+}
+
+// Get the object identifier from an object name
+func (d *driver) GetOid(path string) (string, error) {
+	directory := filepath.Dir(path)
+	base := filepath.Base(path)
+
+	files, err := d.Ioctx.GetOmapValues(directory, "", base, 1)
+
+	if (err != nil) || (files[base] == nil) {
+		return "", storagedriver.PathNotFoundError{Path: path}
+	}
+
+	return string(files[base][:]), nil
+}
+
+// List the objects of a virtual directory
+func (d *driver) ListDirectoryOid(path string) (map[string][]byte, error) {
+	return d.Ioctx.GetOmapValues(path, "", "", defaultMaxKeysFetched)
+}
+
+// Remove a file from the files hierarchy
+func (d *driver) DeleteOid(path string) error {
+	// Remove object reference
+	directory := filepath.Dir(path)
```

Comment:
Considering that we don't want the path layout to change when running on unix or windows, this should use the `path` package rather than `filepath`.